### ISSUE TITLE
[Feature Request] Generate types from @gql annotations

### ISSUE THREAD

**Issue by iscriptology** [comigor/artemis#104]
In relay, one of the strength points is that queries and fragments can be written along side the widgets using them.

We had a discussion regarding this [here](https://github.com/gql-dart/ferry/issues/3).

For example:
```dart

@gql(r"""
  query MyHomePage_query {
    ...AllFilmsList_allFilms
  }
""")
class MyHomePage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return QueryBuilder(
      query: MyHomePage_query(),
      builder: (context, QueryResponse<$MyHomePage_query> response) { 
        // The QueryBuilder implicitly exposese through context the AllFilmsList_allFilms fragment
        return AllFilmsList();
      }
    );
  }
}

@gql(r"""
  fragment AllFilmsList_allFilms {
    allFilms {
      title
    }
  }
""")
class AllFilmsList extends StatelessWidget {
  const AllFilmsList({Key key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return FragmentBuilder<$AllFilmsList_allFilms>(
      builder: (context, allFilms) {

      }
    );
  }
}

```

I also have been able to make vscode to auto-complete me inside the annotation's query.

![](https://camo.githubusercontent.com/c99dbdb5fafa4e4bc25d73cbfd19bc3bbffa47eb/68747470733a2f2f73352e67696679752e636f6d2f696d616765732f657a6769662d332d6433306139663330643864352e676966)

What do you think?

**Comment by comigor** [comment]
Hello @iscriptology!

This seems a nice usage of GraphQL through a Flutter app! I like to use it on relay as well (though one downside is queries end up scattered across the repository and this makes me mad).

However, Artemis is a pure Dart library, and this wouldn't be its responsability to handle: it'd be nice to have a new GraphQL client library that leverages both Artemis type generation and this relay-style annotation-based usage.

If you'd like to start some implementation on this, please reach me and I can assist on the Artemis integration!

**Comment by iscriptology** [comment]
@comigor thanks for the reply.

Even though a specific graphql client adjusted for this use case would be nice, it still is fully backward compatible to just source the graphql queries from the `gql` annotations in addition to the `.graphql` files.

This can be done here at artemis regardless of any client being implemented.

What do you think?

**Comment by comigor** [comment]
Absolutely! We just need to keep configurations working and share fragments/enums/input types correctly.

But to be honest, this implementation is not a priority right now, but I'll gladly assist you if you want to start it!
